containerize.UnitTests/ParserTests.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using Microsoft.NET.Build.Containers;
+using Xunit;
+
+namespace containerize.UnitTests;
+
+public class ParserTests
+{
+    [Fact]
+    public void CanParseLabels()
+    {
+        ContainerizeCommand command = new();
+        DirectoryInfo publishDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), DateTime.Now.ToString("yyyyMMddHHmmssfff"), nameof(CanParseLabels)));
+        List<string> baseArgs = new()
+        {
+            publishDir.FullName,
+            command.BaseRegistryOption.Aliases.First(),
+            "MyBaseRegistry",
+            command.BaseImageNameOption.Aliases.First(),
+            "MyBaseImageName",
+            command.ImageNameOption.Aliases.First(),
+            "MyImageName",
+            command.WorkingDirectoryOption.Aliases.First(),
+            "MyWorkingDirectory",
+            command.EntrypointOption.Aliases.First(),
+            "MyEntryPoint"
+        };
+
+        baseArgs.Add(command.LabelsOption.Aliases.First());
+        baseArgs.Add("NoValue=\"\"");
+        baseArgs.Add("NoValue2=");
+        baseArgs.Add("Valid2=\"Val2\"");
+        baseArgs.Add("Valid3=\"Val 3\"");
+        baseArgs.Add("Valid4=Val4");
+        baseArgs.Add("Unbalanced1=\"Un1");
+        baseArgs.Add("Unbalanced2=Un2\"");
```

Comment:
@baronfel do we need more strict parsing?
i.e. always require quotes etc.

Reply:
Here are the rules for labels and annotations: https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules. The relevant portions are:

> * Annotations MUST be a key-value map where both the key and value MUST be strings.
> * While the value MUST be present, it MAY be an empty string.

So unbalanced quotes are invalid by spec and should be rejected, and `""` should be allowed.

Reply:
note, this applies to containerize CLI only.  We need a format to pass key-value pairs, and task implementation currently sends it in format `key="value"` (note additional quotes), assuming
- key should not have spaces and should not be quoted
- value may have spaces, therefore always quoted.
This part was not changed recently.

This is not 100% aligned to spec as only restriction for them to be string, though they should be namespaces (but not MUST). Will it make more sense to quote whole pair instead as `"key=value"` , split by '=' and trim spaces; and then any quote left will be treated as part of content. 
Quoting part of argument is tricky as the CLI won't help unquote for you, therefore the issues happened before with having extra quotes.

When the image is built outer quotes are removed. In json it would be: ` "key" : "value" ` pair with no additional quotes (here quotes are JSON only).

Reply:
Ah, I see - yes, I think that

> quote whole pair instead as "key=value" , split by '=' and trim spaces; and then any quote left will be treated as part of content

Is a perfectly fine way to go. Thank you for the clarification!

Reply:
Great, then will fix this tomorrow - it should simplify the logic even more.